saleor/giftcard/utils.py

```diff
@@ -53,16 +55,23 @@ def add_gift_card_code_to_checkout(
     checkout.save(update_fields=["last_change"])
 
 
-def remove_gift_card_code_from_checkout(checkout: Checkout, gift_card_code: str):
-    """Remove gift card data from checkout by code.
+def remove_gift_card_code_from_checkout_or_error(
+    checkout: Checkout, gift_card_code: str
+) -> None:
+    """Remove gift card data from checkout by code or raise an error."""
 
-    Return information whether promo code was removed.
-    """
     if gift_card := checkout.gift_cards.filter(code=gift_card_code).first():
         checkout.gift_cards.remove(gift_card)
         checkout.save(update_fields=["last_change"])
-        return True
-    return False
+    else:
+        raise ValidationError(
+            {
+                "promo_code": ValidationError(
+                    "Cannot remove a gift card not attached to this checkout.",
+                    code=CheckoutErrorCode.GIFT_CARD_NOT_APPLICABLE.value,
+                )
+            }
+        )
```

Comment:
You shouldn't wrap the `ValidationError` here with the field name reference, as it might be different for different mutations. You should catch this error in mutation and there add a additional layer with the input field name
```suggestion
    else:
        raise ValidationError(
              "Cannot remove a gift card not attached to this checkout.",
              code=CheckoutErrorCode.GIFT_CARD_NOT_APPLICABLE.value,
        )
```